TTS/tts/models/vits.py

```diff
@@ -878,12 +905,24 @@ def forward(  # pylint: disable=dangerous-default-value
 
         # select a random feature segment for the waveform decoder
         z_slice, slice_ids = rand_segments(z, y_lengths, self.spec_segment_size, let_short_samples=True, pad_short=True)
+
+        if self.args.TTS_part_sample_rate:
+            slice_ids = slice_ids * int(self.interpolate_factor)
+            spec_segment_size = self.spec_segment_size * int(self.interpolate_factor)
+            if self.args.interpolate_z:
+                z_slice = z_slice.unsqueeze(0)  # pylint: disable=not-callable
+                z_slice = torch.nn.functional.interpolate(
+                    z_slice, scale_factor=[1, self.interpolate_factor], mode="nearest"
+                ).squeeze(0)
```

Comment:
Can you make upsampling a separate function and call it here?

Reply:
maybe ```init_upsampling``` to set the arguments and ```upsample``` to run the relevant code.

Reply:
Done